### ISSUE TITLE
Bundle DLLs with patcher in EXE if possible

### ISSUE THREAD

**Issue by lukespragg** [RustOxide/Oxide-2-Patcher#5]

**Comment by HellUnit** [comment]
I've found a way to do this. Basically, just add ILMerge to the solution through the Visual Studio NuGet package manager, then add:
```
"$(ProjectDir)..\packages\ilmerge.2.14.1208\tools\ILMerge.exe" /out:"$(ProjectDir)..\OxidePatcher.exe" "$(TargetPath)" "$(ProjectDir)..\Dependencies\ICSharpCode.TextEditor.dll" "$(ProjectDir)..\Dependencies\Mono.Cecil.dll" "$(ProjectDir)..\Dependencies\Newtonsoft.Json.dll"
```
in the "Build Events\Post Build Event command line" in Project Properties. An all-in-one executable is created.

**Comment by HellUnit** [comment]
I've added a .gitattributes file to specify that *.exe extension is binary and added the ILMerge files to the repository. The modified .csproj file should include the new command line command, so I can PR this stuff if you want. Also modified the .gitignore to ignore the executable and .pdb file created by ILMerge on post-build.

**Comment by lukespragg** [comment]
As long as it works and looks good. Might as well PR.

**Comment by HellUnit** [comment]
Sh*t... I took a look at the license for ILMerge and its not redistributable, so users would have to add it to VS on their own. That's a ball breaker. Thanks MS...

**Comment by lukespragg** [comment]
An issue I see with ILMerge: "Currently, ILMerge works only on Windows-based platforms. It does not yet support Rotor or Mono."

Our builds are created via Mono on Linux.

**Comment by HellUnit** [comment]
There are other ways to do it from what I saw, but none are quite as straightforward.

**Comment by HellUnit** [comment]
Alright, finally got it. It was actually really easy. Add dependencies to project, change file Build Action to "Embedded Resource", then register a callback method with the AppDomain’s ResolveAssembly event in main(). Outlined [url=http://blogs.msdn.com/b/microsoft_press/archive/2010/02/03/jeffrey-richter-excerpt-2-from-clr-via-c-third-edition.aspx]here[/url]. I had to modify it a little to get it to work, but I've got it working now.
